hoodie-hadoop-mr/src/main/java/com/uber/hoodie/hadoop/realtime/AbstractRealtimeRecordReader.java

```diff
@@ -323,13 +340,21 @@ private static Schema addPartitionFields(Schema schema, List<String> partitionin
    * the base split was written.
    */
   private void init() throws IOException {
-    writerSchema = new AvroSchemaConverter().convert(baseFileSchema);
-    List<String> fieldNames = writerSchema.getFields().stream().map(Field::name).collect(Collectors.toList());
+    Schema schemaFromLogFile = null;
+    HoodieTableMetaClient metaClient = new HoodieTableMetaClient(jobConf, split.getBasePath());
+    HoodieTimeline completedTimeline = metaClient.getActiveTimeline().getCommitsTimeline()
```

Comment:
Move this whole logic including readSchemaFromLogFile to a utility class. It can be usable for CLI too.

Reply:
done